src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -485,13 +485,17 @@ public override bool TryGetValue(string key, [NotNullWhen(true)] out string? val
                         {
                             value = key switch
                             {
-                                "indent_style" => _service.UseSpacesForWhitespace(text) ? "space" : "tab",
-                                "tab_width" => _service.GetTabSize(text).ToString(),
-                                "indent_size" => _service.GetIndentSize(text).ToString(),
+                                "indent_style" => _service.UseSpacesForWhitespace(text) switch { true => "space", false => "tab", null => null },
+                                "tab_width" => _service.GetTabSize(text)?.ToString(),
+                                "indent_size" => _service.GetIndentSize(text)?.ToString(),
                                 _ => throw ExceptionUtilities.UnexpectedValue(key)
                             };
 
-                            return true;
+                            // Note that the document might have been closed and the buffer released by the time we get here.
```

Comment:
So I'm surprised this code here is jumping back to the CurrentSolution here in the first place -- since you're now handling the case where the document might not come from a text buffer, why not just have this AnalyzerConfigWithInferredIndentationOptions hold onto DocumentState in the first place? Because in the case where this was now closed, you could have still analyzed the original text that came from a snapshot, no?

Reply:
(put another way: now that the _workspace.IsDocumentOpen check has proven not to work, can you just get rid of it?)